### PR TITLE
fix: Allow running unit tests on macOS when using debug_build_xcframework.sh

### PR DESCRIPTION
**Pull request by Johennes**
This is a small stopgap until #1023 lands to allow running the unit tests for the Swift bindings on macOS. In a nutshell, you can now run this to compile the bindings and execute the tests:

```
$ PLATFORM=darwin ./debug_build_xcframework.sh
$ swift test
```

I also fixed the documentation which was still referring to the old Xcode project instead of the Swift package.

Fixes: #1077

### CODE REVIEW

**Comment by Johennes** [comment]
@stefanceriu / @pixlwave: I had opened this as a stopgap to fix #1077. Being able to _quickly_ run the unit tests on macOS seemed helpful to me. I used it when working on #1078, for instance. However, since #1077 was closed, maybe this one can be discarded, too?

**Comment by stefanceriu** [comment]
The plan is to replace the `build_xcframework` scripts with `xtask` ones and make it run on macOS by default. For now though I don't see why we can't merge this PR as it is. What do you think @pixlwave?

**Comment by pixlwave** [comment]
Yep yep, sounds like a good stop gap until we're back onto EX 👍

**Review by pixlwave — APPROVED** [review]
Not sure if you need a ✅ from one of us, but just in case.

**Comment by Johennes** [comment]
Ah, if you're ok I'd need you to merge it actually. I don't have merge rights here.

**Comment by Johennes** [comment]
@stefanceriu and I debated the (black) bash magic in DM and decided to discard this over #1023.

**Comment by pixlwave** [comment]
@Johennes It would be great to not lose your updated docs from this PR though (either as part of #1023 or a follow-up)

**Comment by Johennes** [comment]
Ah good point. I'll try and salvage that tonight.

**Pull request closed without merging**
